code/modules/paperwork/faxmachine.dm

```diff
@@ -0,0 +1,58 @@
+
+//The IA Agent's Fax Machine
+
+/obj/machinery/faxmachine
+	name = "fax machine"
+	desc = "Used to send messages to your corporate overlords (and promptly get fired for writing fanfiction about their mothers)."
+	circuit = /obj/item/circuitboard/machine/faxmachine
+	icon = 'icons/obj/library.dmi'
+	icon_state = "fax"
+	density = TRUE
+	idle_power_usage = 30
+	active_power_usage = 200
+	COOLDOWN_DECLARE(static/important_action_cooldown)
+
+/obj/machinery/faxmachine/attack_hand(mob/living/user)
+	. = ..()
+	add_fingerprint(user)
+	if(!COOLDOWN_FINISHED(src, important_action_cooldown))
+		to_chat(user, "<span class='warning'>The subspace communications transmissions system is on cooldown!</span>")
+		return
+	if(panel_open)
+		return
+
+	var/associates = (obj_flags & EMAGGED) ? "the Syndicate" : "Central Command"
+	var/msg = capped_multiline_input(user, "Enter a message to be faxed to [associates]", "Fax machine", )
```

Comment:
You check the cooldown before the user input, and set it afterwards meaning that someone can open up multiple dialogue boxes (since the cooldown will be ready), then send hundreds of messages at once (Even though the cooldown is active, they can still send the messages as long as they opened the dialogue before the cooldown was set).
Any checks that happen before a proc that can wait needs to happen aftewards too.

Reply:
fixed, message wont be send with additional inputs in rapid succession